provider/ec2/environ.go

```diff
@@ -177,6 +182,20 @@ func (e *environ) Create(ctx context.ProviderCallContext, args environs.CreatePa
 
 // Bootstrap is part of the Environ interface.
 func (e *environ) Bootstrap(ctx environs.BootstrapContext, callCtx context.ProviderCallContext, args environs.BootstrapParams) (*environs.BootstrapResult, error) {
+	// We are going to take a look at the Bootstrap constraints and see if we have to make an instance profile
+	if args.BootstrapConstraints.HasInstanceRole() &&
+		*args.BootstrapConstraints.InstanceRole == awsInstanceProfileAutoCreateVal {
+
+		controllerName, ok := args.ControllerConfig[controller.ControllerName].(string)
+		if !ok {
+			return nil, errors.NewNotFound(nil, "cannot find controller name in config")
```

Comment:
Not sure that this should be NotFound.
I'd use NotValid as it's an invalid config not to have the controller name.